src/lib.rs

```diff
@@ -234,6 +234,12 @@ impl<T> Event<T> {
     /// The default implementation (implemented on `usize`) is a notification that only notifies
     /// *at least* the specified number of listeners.
     ///
+    /// This function returns the number of [`EventListener`]s that were notified by this
+    /// call. However, in some cases, the notification will be delayed and this function will
+    /// return `0` instead of the number of listeners actually notified. Therefore, the return
+    /// value of this function should not be relied upon for soundness and should be used only
+    /// as a hint.
```

Comment:
This sounds very fishy to me and will basically just mean "I best not use this" if they can't rely on it. At the very least, we should specify the conditions under which the value is unreliable so users have some clue of when they can and can not rely on it.

Reply:
I've specified how it works under the hood now